### ISSUE TITLE
[BUG] Expect() with message arg is shown inside and outside of test.step in trace viewer

### ISSUE THREAD

**Issue by davidjonesdialexa** [microsoft/playwright#26715]
### System info
- Playwright Version: [v1.37]
- Operating System: [Mac OS 12.6.1]
- Browser: [Chromium]
- Other info:

![Screen Shot 2023-08-25 at 4 41 25 PM](https://github.com/microsoft/playwright/assets/84142414/e0ac5397-321c-4c6b-bb4c-35d913493826)

### Source code

- [x] I provided exact source code that allows reproducing the issue locally.

**Config file**

```js
// @ts-check
const { defineConfig, devices } = require('@playwright/test');

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * @see https://playwright.dev/docs/test-configuration
 */
module.exports = defineConfig({
  testDir: './tests',
  /* Run tests in files in parallel */
  fullyParallel: true,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  reporter: 'html',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
    /* Base URL to use in actions like `await page.goto('/')`. */
    // baseURL: 'http://127.0.0.1:3000',

    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on',
    actionTimeout: 3000,
    navigationTimeout: 10000,
  },

  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'], channel: 'chrome' },
    },

    // {
    //   name: 'firefox',
    //   use: { ...devices['Desktop Firefox'] },
    // },

    // {
    //   name: 'webkit',
    //   use: { ...devices['Desktop Safari'] },
    // },

    /* Test against mobile viewports. */
    // {
    //   name: 'Mobile Chrome',
    //   use: { ...devices['Pixel 5'] },
    // },
    // {
    //   name: 'Mobile Safari',
    //   use: { ...devices['iPhone 12'] },
    // },

    /* Test against branded browsers. */
    // {
    //   name: 'Microsoft Edge',
    //   use: { ...devices['Desktop Edge'], channel: 'msedge' },
    // },
    // {
    //   name: 'Google Chrome',
    //   use: { ..devices['Desktop Chrome'], channel: 'chrome' },
    // },
  ],

  /* Run your local dev server before starting the tests */
  // webServer: {
  //   command: 'npm run start',
  //   url: 'http://127.0.0.1:3000',
  //   reuseExistingServer: !process.env.CI,
  // },
});

```

**Test file (self-contained)**

```js
test('hotwire', async ({ page }) => {
  await page.goto('https://www.hotwire.com/');
  await test.step('expect find a hotel', async () => {
    await expect(
      page.getByRole('button', { name: 'Find a hotel' }),
      'expect to have text: find a hotel'
    ).toHaveText('Find a hotel');
  });
});
```

**Steps**
- [Run the test]
- [view the reporter]

**Expected**

in HTML reporter, action, represented by the message passed to expect() displayed within the test.step

**Actual**

in HTML reporter, action, represented by the message passed to expect() is displayed within the test.step, but the expect call (expect.toHaveText) is also shown in the HTML reporter outside of the test.step

**Comment by yury-s** [comment]
I cannot reproduce it, this is what I get with playwright 1.37.1: 

![image](https://github.com/microsoft/playwright/assets/9798949/6dfdf19d-fb3c-4e70-b304-3438be48ebd9)

**Comment by pavelfeldman** [comment]
The user means trace viewer, not html report.

**Comment by davidjonesdialexa** [comment]
Sorry about that, yes I meant the trace viewer

**Comment by dgozman** [comment]
Fixed by #26752.
